source/proxy_connection.c

```diff
@@ -1204,6 +1204,25 @@ static int s_connect_proxy(const struct aws_http_client_connection_options *opti
         return AWS_OP_ERR;
     }
 
+    if (options->proxy_options->no_proxy_hosts.len > 0) {
+        struct aws_string *no_proxy_host_str = aws_string_new_from_cursor(options->allocator, &options->proxy_options->no_proxy_hosts);
+        if (aws_http_host_matches_no_proxy(options->allocator, options->host_name, no_proxy_host_str)) {
+            AWS_LOGF_DEBUG(
+                AWS_LS_HTTP_CONNECTION,
+                "Host \"" PRInSTR "\" found in NO_PROXY, bypassing proxy",
+                AWS_BYTE_CURSOR_PRI(options->host_name));
```

Comment:
Trivial: let's log the no_proxy str as well, so that easier to debug.
```suggestion
            AWS_LOGF_DEBUG(
                AWS_LS_HTTP_CONNECTION,
                "Host \"" PRInSTR "\" found in no_proxy_hosts: \" %s \", bypassing proxy",
                AWS_BYTE_CURSOR_PRI(options->host_name),
                no_proxy_host_str->ptr);
```